.github/actions/start-opensearch/action.yml

```diff
@@ -0,0 +1,175 @@
+    required: true
+
+  plugin-zip:
+    description: 'The name of the zip file for the plugin hosted on docker-host i.e. security-plugin.zip '
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+
+    # Configure longpath names if on Windows
+    - name: Enable Longpaths if on Windows
+      if: ${{ inputs.platform == 'windows-latest' }}
+      run: git config --system core.longpaths true
+      shell: pwsh
+
+    # Downloads the min version of OpenSearch Core from the latest build
+    - name: Download OpenSearch Core for Linux
+      if: ${{ inputs.platform == 'ubuntu-latest' }}
+      run: |
+        wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-${{ inputs.opensearch-version }}-linux-x64.tar.gz
+        tar -xzf opensearch-*.tar.gz
+        rm -f opensearch-*.tar.gz
+      shell: bash
+
+    - name: Download OpenSearch Core for Windows
+      if: ${{ inputs.platform == 'windows-latest' }}
+      run: |
+        Invoke-WebRequest https://artifacts.opensearch.org/snapshots/core/opensearch/${{ inputs.opensearch-version }}-SNAPSHOT/opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-windows-x64-latest.zip -Outfile opensearch-${{ inputs.opensearch-version }}.zip
+        tar -xzf opensearch-${{ inputs.opensearch-version }}.zip
+        del opensearch-${{ inputs.opensearch-version }}.zip
+        Rename-Item opensearch-* opensearch
+      shell: pwsh
+
+    - name: Download OpenSearch Core for MacOS
+      if: ${{ inputs.platform == 'macos-latest' }}
+      run: |
+        curl https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-${{ inputs.opensearch-version }}-linux-x64.tar.gz
+        tar -xzf opensearch-*.tar.gz
+        rm -f opensearch-*.tar.gz
+        mv opensearch-* opensearch
+      shell: bash
+
+    # Assemble the specified plugin
+    - name: Assemble the specified plugin for MacOS and Linux
+      if: ${{ inputs.platform != 'windows-latest' }}
+      run: ./gradlew clean assemble
+      shell: bash
+
+    - name: Assemble the specified plugin for Windows
+      if: ${{ inputs.platform == 'windows-latest' }}
+      run: .\gradlew.bat assemble
+      env:
+        _JAVA_OPTIONS: -Xmx4096M
+      shell: pwsh
+
+    # Move and rename the plugin for installation
+    - name: Move and rename the plugin for installation on MacOS and Linux
+      if: ${{ inputs.platform != 'windows-latest' }}
+      run: mv build/distributions/${{ inputs.plugin-name }}-*.zip ${{ inputs.plugin-name }}.zip
+      shell: bash
+
+    - name: Move and rename the plugin for installation on Windows
+      if: ${{ inputs.platform == 'windows-latest' }}
+      run: move build\distributions\${{ inputs.plugin-name }}-*.zip .\${{ inputs.plugin-name }}.zip
+      shell: pwsh
```

Comment:
I think we can reuse the same action for both platforms, we don't need to run on `pwsh` if `shell` works.